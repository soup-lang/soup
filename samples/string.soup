func ret_string()
    return "fla" + 'tscript'
stekin: ret_string()
console.log(stekin)

console.log(stekin + true)
stekin2: stekin + stekin
console.log(stekin2)

x: 112
console.log(x + stekin)

console.log(3.14 + "" + 3.14)

func add_str(s)
    func adding(a)
        return s + a
    return adding

y: add_str('abc
    ')
console.log(y(10))
console.log(y(false))

console.log('abcabc'.replace(*RegExp('b', 'g'), 'mm'))

console.log(::: left:unindent eol:keep
    The quick brown fox
        jumps over a lazy dog
::: + '.')
