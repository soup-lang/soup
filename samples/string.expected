soup
souptrue
soupsoup
112soup
3.143.14
abc
    10
abc
    false
ammcammc
The quick brown fox
    jumps over a lazy dog.
